msrest/serialization.py

```diff
@@ -552,13 +546,14 @@ def body(self, data, data_type, **kwargs):
         # Just in case this is a dict
         internal_data_type = data_type.strip('[]{}')
         internal_data_type = self.dependencies.get(internal_data_type, None)
+        is_xml_model_serialization = kwargs.get("is_xml", False)
```

Comment:
Should be the same logic is_xml_model_serialization = kwargs.setdefault("is_xml", target_obj.is_xml_model())